rust/core-lib/src/plugins/rpc.rs

```diff
@@ -183,7 +185,106 @@ pub enum PluginNotification {
     Alert { msg: String },
     AddStatusItem { key: String, value: String, alignment: String },
     UpdateStatusItem { key: String, value: String  },
-    RemoveStatusItem { key: String }
+    RemoveStatusItem { key: String },
+    ShowHover { request_id: usize, result: Result<Hover, LanguageResponseError>, rev: u64 },
+    ShowDefinition { request_id: usize, result: Result<Definition, LanguageResponseError>, rev: u64}
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+#[serde(rename_all = "snake_case")]
+#[serde(tag = "type")]
+pub enum Position {
+    Utf8Offset { offset: usize },
+    Utf16LineChar { line: usize, character: usize},
+    Utf8LineChar { line: usize, character: usize }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+pub struct Range {
+    pub start: Position,
+    pub end: Position
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum LanguageResponseError {
+    LanguageServerError(String),
+    PositionConversionError(String),
+    NullResponse,
+    FallbackResponse
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+pub struct Hover {
+    pub content: String,
+    pub range: Option<Range>
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+pub struct Definition {
+    pub locations: Vec<Location> 
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+pub struct Location {
+    pub path: PathBuf,
+    pub range: Range
+}
+
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+pub struct CompletionItem {
+    pub label: String,
+    pub kind: Option<CompletionItemKind>,
+    pub detail: Option<String>,
+    pub documentation: Option<String>,
+    pub sort_text: Option<String>,
+    pub filter_text: Option<String>,
+    pub primary_edit: TextEdit,
+    pub secondary_edits: Option<Vec<TextEdit>>
+}
+
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+pub struct TextEdit {
```

Comment:
The idea was to receive such a TextEdit from Plugin and convert it to Delta in the Core, since it involves the same problem like converting Utf-16 Line/Char Range to Utf8 offsets in the Plugin, so I resorted to forwarding it to core.